github/github.go

```diff
@@ -895,12 +902,24 @@ type BasicAuthTransport struct {
 
 // RoundTrip implements the RoundTripper interface.
 func (t *BasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	req = cloneRequest(req) // per RoundTrip contract
-	req.SetBasicAuth(t.Username, t.Password)
+	// To set extra headers, we must make a copy of the Request so
+	// that we don't modify the Request we were given. This is required by the
+	// specification of http.RoundTripper.
+	//
+	// Since we are going to modify only req.Header here, we only need a deep copy
+	// of req.Header.
+	req2 := new(http.Request)
+	*req2 = *req
+	req2.Header = make(http.Header, len(req.Header))
+	for k, s := range req.Header {
+		req2.Header[k] = append([]string(nil), s...)
```

Comment:
Why not `[]string{}` instead of `[]string(nil)`?
I see it was written that way before in `cloneRequest` but I'm not sure I understand the difference.
`[]string{}` seems more common (to me) but maybe it's not a big deal?

Reply:
@gmlewis good one! I didn't think about it. Definitely, `[]string{}` is more common one. I'll change that in next commit. Thanks for reviewing!

Reply:
> Why not `[]string{}` instead of `[]string(nil)`?

@gmlewis I think it's because that's the style used in [Go slice tricks document](https://github.com/golang/go/wiki/SliceTricks#copy), where this "copy slice" pattern is taken from. I think it's better to stay consistent with that.

If you think it should be changed, I suggest you make the proposal to change Go slice tricks, then we can update our style here.

Reply:
Oh, weird!  Thanks for the pointer, @shurcooL. OK, I'm fine with keeping it that way.  I just thought it was odd-looking... looks odd there too, but that's fine... no need to change it. I'll shut up now.  :smile:

Reply:
Thanks for the pointer, @shurcooL. Changed it back 😄